internal/app/init/pkg/network/network.go

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package network
+
+import (
+	"log"
+	"time"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv4/client4"
+	"github.com/insomniacslk/dhcp/netboot"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+func Setup(name string) (err error) {
+	link, err := netlink.LinkByName("lo")
+	if err != nil {
+		return err
+	}
+	if err = netlink.LinkSetUp(link); err != nil {
+		return err
+	}
+
+	ifname := "eth0"
+	if _, err = netboot.IfUp(ifname, 5*time.Second); err != nil {
+		return err
+	}
+	modifiers := []dhcpv4.Modifier{}
+	switch name {
+	case "GoogleCloud":
+		modifiers = append(modifiers, dhcpv4.WithRequestedOptions(dhcpv4.OptionClasslessStaticRouteOption))
+	}
+	var netconf *netboot.NetConf
+	if netconf, err = dhclient4(ifname, modifiers...); err != nil {
+		return err
+	}
+	if err = netboot.ConfigureInterface(ifname, netconf); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func dhclient4(ifname string, modifiers ...dhcpv4.Modifier) (*netboot.NetConf, error) {
+	attempts := 5
+	client := client4.NewClient()
+	var (
+		conv []*dhcpv4.DHCPv4
+		err  error
+	)
+	for attempt := 0; attempt < attempts; attempt++ {
+		log.Printf("attempt %d of %d for DHCP lease", attempt+1, attempts)
+		conv, err = client.Exchange(ifname, modifiers...)
+		if err != nil && attempt < attempts {
+			log.Printf("failed to request DHCP lease: %v", err)
```

Comment:
Might be worth adding in a real dumb sleep/sleep w/ backoff here.

Reply:
agree 👍